core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java

```diff
@@ -53,6 +55,21 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
     this.name = name;
     this.parentname = parentname;
     this.isDirectDictionary = isDirectDictionary;
+    this.isDictionary = true;
+  }
+
+
+  public PrimitiveQueryType(String name, String parentname, int blockIndex,
```

Comment:
One more extra constructor is not required. If dictionary is not null then isDictionary=true

Reply:
Done. But we need the parameter as dictionary field is always set.

Reply:
Moved into a single constructor. But we cannot check is the column is a dictionary or nodictionary through the existing variable dictionary as it is always going to get filled up by default.